dispatch/route.go

```diff
@@ -180,6 +180,30 @@ func (r *Route) Key() string {
 	return b.String()
 }
 
+// ID returns a key for the route. It should uniquely identify the route in general,
+// it is different than Key() as it adds the route's position on its parent's children.
+func (r *Route) ID() string {
+	b := strings.Builder{}
+
+	var position *int
```

Comment:
I prefer to not use pointer... We can use int and initialize it to an invalid value like `-1` then we will have less pointer reference operations

Reply:
fixed